File: app/src/main/java/za/co/riggaroo/datecountdown/entity/Event.java
package za.co.riggaroo.datecountdown.entity;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import org.threeten.bp.LocalDateTime;
import org.threeten.bp.temporal.ChronoUnit;

import static za.co.riggaroo.datecountdown.entity.Event.TABLE_NAME;

@Entity(tableName = TABLE_NAME)
public class Event {
    public static final String TABLE_NAME = "events";
    public static final String DATE_FIELD = "date";

    @PrimaryKey(autoGenerate = true)
    private int id;
    private String name;
    private String description;
    @ColumnInfo(name = DATE_FIELD)
    private LocalDateTime date;

    public Event(int id, String name, String description, LocalDateTime date) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.date = date;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public LocalDateTime getDate() {
        return date;
    }

    @Override
    public String toString() {
        return "Event{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", date=" + date +
                '}';
    }

    public long getDaysUntil() {
        return ChronoUnit.DAYS.between(LocalDateTime.now(), getDate());
    }
}

File: app/src/main/java/za/co/riggaroo/datecountdown/data/entity/Event.kt
package za.co.riggaroo.datecountdown.data.entity

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import org.threeten.bp.LocalDateTime
import org.threeten.bp.temporal.ChronoUnit
import za.co.riggaroo.datecountdown.data.entity.Event.Companion.TABLE_NAME


@Entity(tableName = TABLE_NAME)
data class Event(@PrimaryKey(autoGenerate = true)
            val id: Int,
            val name: String,
            val description: String,
            @ColumnInfo(name = DATE_FIELD)
            val date: LocalDateTime) {

    fun daysUntil(): Long {
        return ChronoUnit.DAYS.between(LocalDateTime.now(), date)
    }

    companion object {
       const val TABLE_NAME = "events"
       const val DATE_FIELD = "date"
    }
}
